src/path_tracking_pid_local_planner.cpp

```diff
@@ -0,0 +1,644 @@
+  costmap_ = costmap;
+  tf_ = tf;
+
+  initialized_ = true;
+}
+
+bool TrackingPidLocalPlanner::setPlan(const std::vector<geometry_msgs::PoseStamped>& global_plan)
+{
+  if (!initialized_)
+  {
+    ROS_ERROR("path_tracking_pid has not been initialized, please call initialize() before using this planner");
+    return false;
+  }
+
+  std::string path_frame = global_plan.at(0).header.frame_id;
+  ROS_DEBUG("TrackingPidLocalPlanner::setPlan(%zu)", global_plan.size());
+  ROS_DEBUG("Plan is defined in frame '%s'", path_frame.c_str());
+
+  global_plan_ = global_plan;
+
+  /* If frame of received plan is not equal to mbf-map_frame, translate first */
+  if (map_frame_.compare(path_frame))
+  {
+    ROS_DEBUG("Transforming plan since my global_frame = '%s' and my plan is in frame: '%s'", map_frame_.c_str(),
+              path_frame.c_str());
+    geometry_msgs::TransformStamped tf_transform;
+    tf_transform = tf_->lookupTransform(map_frame_, path_frame, ros::Time(0));
+    // Check alignment, when path-frame is severly mis-aligned show error
+    double yaw, pitch, roll;
+    tf2::getEulerYPR(tf_transform.transform.rotation, yaw, pitch, roll);
+    if (std::fabs(pitch) > MAP_PARALLEL_THRESH || std::fabs(roll) > MAP_PARALLEL_THRESH)
+    {
+      ROS_ERROR("Path is given in %s frame which is severly mis-aligned with our map-frame: %s", path_frame.c_str(),
+                                                                                                 map_frame_.c_str());
+    }
+    for (auto& pose_stamped : global_plan_)
+    {
+      tf2::doTransform(pose_stamped.pose, pose_stamped.pose, tf_transform);
+      pose_stamped.header.frame_id = map_frame_;
+      // 'Project' plan by removing z-component
+      pose_stamped.pose.position.z = 0.0;
+    }
+  }
+
+  if (controller_debug_enabled_)
+  {
+    received_path_.header = global_plan_.at(0).header;
+    received_path_.poses = global_plan_;
+    path_pub_.publish(received_path_);
+  }
+
+  try
+  {
+    ROS_DEBUG("map_frame: %s, plan_frame: %s, base_link_frame: %s", map_frame_.c_str(), path_frame.c_str(),
+              base_link_frame_.c_str());
+    tfCurPoseStamped_ = tf_->lookupTransform(map_frame_, base_link_frame_, ros::Time(0));
+  }
+  catch (tf2::TransformException ex)
+  {
+    ROS_ERROR("Received an exception trying to transform: %s", ex.what());
+    return false;
+  }
+
+  // Feasability check, but only when not resuming with odom-vel
+  if (pid_controller_.getConfig().init_vel_method != Pid_Odom &&
+      pid_controller_.getConfig().init_vel_max_diff >= 0.0 &&
+      std::abs(latest_odom_.twist.twist.linear.x - pid_controller_.getControllerState().current_x_vel) >
+        pid_controller_.getConfig().init_vel_max_diff)
+  {
+    ROS_ERROR("Significant diff between odom (%f) and controller_state (%f) detected. Aborting!",
+              latest_odom_.twist.twist.linear.x, pid_controller_.getControllerState().current_x_vel);
+    return false;
```

Comment:
I think this was introduced when an user cancels an ongoing action and moves the robot manually. Then wen restarted, there might be a big difference between controller internal setpoint state and the current velocity.

Reply:
Sounds like a reason to reset the controller state, not to reject the goal. This is something you can always do on reception of a new goal, right? Simply by using the current state of the robot.

Reply:
This check is only done in one of the operating modes. Documentation, variable names and error messages, needs to be updated

Reply:
Another idea is to abort the execution at the controller step function. This makes that the odometry subscription is not needed anymore because move_base_flex provides it when the controller is active.

Reply:
See #94